bin/picca_Pk1D.py

```diff
@@ -7,152 +7,367 @@
+            "Could not figure out if linear or log wavelength binning was used, probably submitted lambda as log_lambda"
+        )
+    else:
+        raise ValueError(
+            "Could not figure out if linear or log wavelength binning was used"
+        )
+
+    return linear_binning, delta_lam
+
+
+# loop over input files
+def process_all_files(index_file_args):
+    global num_data
+    file_index, file, args = index_file_args
+    if file_index % 5 == 0:
+        userprint("\rread {} of {} {}".format(file_index, args.len_files,
+                                              num_data),
+                  end="")
+
+    # read fits or ascii file
+    if args.in_format == 'fits':
+        hdul = fitsio.FITS(file)
+        deltas = [Delta.from_fitsio(hdu, pk1d_type=True) for hdu in hdul[1:]]
+    elif args.in_format == 'ascii':
+        ascii_file = open(file, 'r')
+        deltas = [Delta.from_ascii(line) for line in ascii_file]
+
+    #add the check for linear binning on first spectrum only (assuming homogeneity within the file)
+    delta = deltas[0]
+    linear_binning, delta_lam = check_linear_binning(delta)
+    if linear_binning:
+        userprint("\n\nUsing linear binning, results will have units of AA")
+        delta_lambda = delta_lam
+        if (args.disable_reso_matrix or not hasattr(delta, 'resolution_matrix')
+                or delta.resolution_matrix is None):
+            userprint(
+                "Resolution matrix not found or disabled, using Gaussian resolution correction\n"
+            )
+            reso_correction = "Gaussian"
+        else:
+            userprint("Using Resolution matrix for resolution correction\n")
+            reso_correction = "matrix"
+    else:
+        userprint("\n\nUsing log binning, results will have units of km/s")
+        delta_log_lambda = delta_lam
+        reso_correction = "Gaussian"
+        userprint("Using Gaussian resolution correction\n")
+
+    num_data += len(deltas)
+    userprint("\n ndata =  ", num_data)
+    results = None
+
+    def process_file(delta):
+        """_summary_
+
+        Args:
+            delta (Delta): a single delta object
+
+        Returns:
+            list containing the following for each spectral part
+            k: mode
+            pk_raw: raw power spectrum
+            pk_noise: noise power spectrum from pipeline noise
+            pk_diff: noise power spectrum from difference spectra
+            correction_reso: resolution correction
+            pk: final power spectrum measurement
+            mean_z_array[part_index]: mean redshift within the spectral part
+            num_masked_pixels: number of masked pixels within the spectral part
+        """
+        # Selection over the SNR and the resolution
+        if (delta.mean_snr <= args.SNR_min
+                or delta.mean_reso >= args.reso_max):
+            return None
+
+        # first pixel in forest
+        selected_pixels = 10**delta.log_lambda > args.lambda_obs_min
```

Comment:
the filter for both minimal and maximal obs wavelength is actually done in delta_extraction already, this is mostly to avoid e.g. the very noisy blue end of the spectra even if we computed a continuum for it, on the red end there is no such feature (at least not for any reasonable Lya forest redshift range and eBOSS or DESI).

Regarding the other question: at least at the moment the delta object when read from file will always contain `log_lambda` and for this operation the binning is unimportant. The splits later are due to functions that only work on uniform binning (sometimes except for masking), so there binning needs to be tracked more closely. Generally it might be good to convert this routine to a more elegant modular one, but first it needs to work at all (on the new delta outputs)...

Reply:
About the filtering, maybe it would be better to add a comment saying that filtering is already done and that here we just filter more aggressively on the blue end.

Ok I see for the second question. A couple comments on "binning needs to be uniform" should be added to the relevant parts, so that we keep track of it in the future

Reply:
there are some comments now